addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoBridgeHandler.java

```diff
@@ -119,23 +120,32 @@ private void scheduleTokenInitAndRefresh() {
                 getPartnerApi().partnerdevices();
                 connectionSucceed();
             } catch (RetrofitError e) {
-                switch (e.getResponse().getStatus()) {
-                    case 404: // If no partner station has been associated - likely to happen - we'll have this error
-                              // but it means connection to API is OK
-                        connectionSucceed();
-                        break;
-                    case 403: // Forbidden Access maybe too many requests ? Let's wait next cycle
-                        logger.warn("Error 403 while connecting to Netatmo API, will retry in {} s",
-                                configuration.reconnectInterval);
-                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_REGISTERING_ERROR,
-                                "Netatmo Access Forbidden, will retry in " + configuration.reconnectInterval.toString()
-                                        + " seconds.");
-                        break;
-                    default:
-                        logger.error("Unable to connect Netatmo API : {}", e.getMessage(), e);
-                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                                "Unable to connect Netatmo API : " + e.getLocalizedMessage());
-                        return;
+                if (e.getKind() == Kind.NETWORK) {
+                    logger.warn("Network error while connecting to Netatmo API, will retry in {} s",
+                            configuration.reconnectInterval);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                            "Netatmo Access Failed, will retry in " + configuration.reconnectInterval.toString()
+                                    + " seconds.");
+                } else {
+                    switch (e.getResponse().getStatus()) {
+                        case 404: // If no partner station has been associated - likely to happen - we'll have this
+                                  // error
+                                  // but it means connection to API is OK
+                            connectionSucceed();
+                            break;
+                        case 403: // Forbidden Access maybe too many requests ? Let's wait next cycle
+                            logger.warn("Error 403 while connecting to Netatmo API, will retry in {} s",
+                                    configuration.reconnectInterval);
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_REGISTERING_ERROR,
```

Comment:
Are you sure these thingstatusDetail are correct, @htreu wdyt?

Reply:
Note that this part of code is not mine. That's true that I was a little surprised to see that detailed status.

Reply:
Unfortunately `ThingStatusDetail.HANDLER_REGISTERING_ERROR` has no JavaDoc but from the context it seems to be wrong here. I suggest to go with `ThingStatusDetail .COMMUNICATION_ERROR` along with the already provided description.